docs/guides/run_a_job.md

```diff
@@ -0,0 +1,227 @@
+
+Refer to [this section](../faqs/plans/#preemptible-vs-dedicated-instances)
+
+
+### Dataset(s)
+
+You can specify up to five datasources (datasets or outputs from previous jobs)
+to mount to the server that will be running your job. For each datasource,
+specify the `--data` flag as detailed below:
+
+```
+--data <name_of_datasource>:<mount_point_on_server>
+```
+
+For more detailed information on mounting data to jobs, see
+[this article](../guides/data/mounting_data)
+
+### Mode
+
+FloydHub jobs can currently be run in one of three modes:
+
+1. `--mode job`
+2. `--mode jupyter`
+3. `--mode serve`
+
+Here is a description of each mode:
+
+#### `--mode job`
+
+This is the default mode so there is no need to specify `--mode job` when
+running `floyd run`. You can think of this mode as "regular mode". When you run
+your job in this mode, your code is sent up to a FloydHub deep-learning server
+and the [[COMMAND]](#command) portion of `floyd run` executed.
+
+#### `--mode jupyter`
+
+This is Jupyter Notebook mode. When you specify `--mode jupyter`, your code is
+uploaded to a FloydHub deep-learning server, and a Jupyter Notebook session is
+started in the directory containing your code.
+
+FloyHub serves this Jupyter Notebook session for you. Your Jupyter Notebook
+session will be automatically opened in a web browser when it becomes
+available. If you don't want your browser to automatcially open, pass the `--no-open` flag to `floyd run`.
+
+!!! important "Use This Mode When:"
+    1. You want to work with a **Jupyter Notebook**.
+    2. You want to open an **interactive shell** on the FloydHub server where your code
+       is running. For details on how to do that, see [this article](../guides/ssh)
+
+#### `--mode serve`
+
+This mode is for serving your deep learning models as an API endpoint. To use
+serve mode, you'll need to make sure your project's code meets a few
+requirements:
+
+- Contains a file called `app.py` that contains a Flask application that will
+  listen on port 5000.
+- Contains a `requirements.txt` file that contains the line `flask`, which
+  declares it as a dependency.
+
+!!! warning
+    This mode is currently in development, and not appropriate for production
+    use.
+
+### Environment
+
+Specifying the environment means choosing what major deep-learning software
+packages you want available on the server that runs your code. This is not a
+specification between a CPU server and a GPU server (that's the
+[Instance Type](#instance-type) `OPTION` of `floyd run`).
+
+FloydHub offers servers with many different deep-learning software packages
+pre-installed.  You can find a list of all the available environments
+[here](../guides/environments).
+
+Use the `--env` flag to specify which environment you would like your job to
+run in. It is best practice to pass the entire name of the environment,
+including the version number, to the `--env` flag. For example, instead of
+`--env tensorflow`, use `--env tensorflow-1.3`.
+
```

Comment:
- A couple more examples of different envs here would be useful
- If no version is specified, the latest available version at that time is used. E.g. `--env tensorflow` will use tensorflow-1.3 today